docs/design/configs.md

```diff
@@ -0,0 +1,130 @@
+# Generic Config Object
+
+## Abstract
+
+A generic ``Config`` object to store command-line flags and environment variables
+that is used for configuration of ``Kubernetes`` components living on top of ``etcd``
+is proposed in this proposal.
```

Comment:
If we did a string, it's a bit more like secrets, and also handles things like Apache conf, or redis.conf, or my.cnf... Etc

> On Jun 19, 2015, at 7:01 PM, Tim Hockin <notifications@github.com> wrote:
> 
> In docs/design/configs.md:
> 
> > @@ -0,0 +1,130 @@
> > +# Generic Config Object
> > +
> > +## Abstract
> > +
> > +A generic ``Config`` object to store command-line flags and environment variables
> > +that is used for configuration of ``Kubernetes`` components living on top of ``etcd``
> > +is proposed in this proposal.
> I think we want to discourage people from thinking of this as "just" a replacement for flags. Things like ordering and repeats being prime reasons why. I could maybe buy changing it to a list of k/v pairs to preserve ordering - MAYBE. But repeats are on users. We could spec a string format and offer libs to parse that format into lists, I guess, but I'd just as soon say "use JSON syntax" and then they can parse that. ``` "params" : { "frobbers-supported": "[ \"foo\", \"bar\" ]" } ``` And if we support lists, do we support maps? at some point it becomes a single opaque JSON blob. I think we can start with a simple middle ground - key-value pairs, where values are opaque. Most params will fit this just fine. Or we can go simpler and just say it's a string. Insert JSON here and parse it yourself. As for validation and defaults, I don't think we want that here, but if we do we can grow something like schema - once it is demonstrated. Honestly, I am as anxious as anyone about overly-loose params, but I'd much much rather have something that handles simple cases NOW.
> …
> —
> Reply to this email directly or view it on GitHub.
>